src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1474,6 +1474,13 @@ private IParameterInitializerOperation CreateBoundParameterEqualsValueOperation(
 
         private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
         {
+            // The C# binder produces an extra block for checked, unchecked, and unsafe statements, but they have no semantics of their own
+            boundBlock = boundBlock switch
+            {
+                { Syntax: CheckedStatementSyntax _, Locals: { IsDefaultOrEmpty: true }, Statements: { Length: 1 } s } => (BoundBlock)s[0],
```

Comment:
>Syntax: CheckedStatementSyntax [](start = 18, length = 30)

I think we are trying to follow the principle that semantic meaning of bound nodes shouldn't depend on their associated syntax. This condition violates the principle.